### ISSUE TITLE
WTaxa: test report

### ISSUE THREAD

**Issue by yroskov** [CatalogueOfLife/testing#28]
ACEF data received on 2021-02-21.

Manuel: Attached you can find the latest version of WTaxa (from late 2020).

Like last time, there are duplicate records that we are working on 
step by step. **You can remove the duplicates, but in each case try to 
keep the record with the lowest ID.**

**Comment by yroskov** [comment]
- [x] Subspecies Assigned, 11 (in the source there is a link to valid name)
- [x] Authorship Contains Taxonomic Note, 18 (some excluded)
- [x] Missing Genus, 22
- [x] Multi Word Epithet, 25

**Comment by yroskov** [comment]
- [ ] Partially Parsable Name, 847
https://data.catalogueoflife.org/catalogue/3/dataset/1039/workbench?facet=rank&facet=issue&facet=status&facet=nomStatus&facet=nameType&facet=field&issue=partially%20parsable%20name&limit=500&offset=0&reverse=false
A. Some of names (e.g. Curculio 470 Zschach, J.J., 1788) need to be blocked. However, software/interface(?) is very unstable with many-names-per-page: decision "block" failed to be applied will all selected names. @mdoering 

![image](https://user-images.githubusercontent.com/28574252/109035318-239f2d00-768e-11eb-83b4-dabc7d0620ff.png)

B. I am not able to remove colons in hundreds of names like that (even search for ":" delivers zero results): 
Charamus: lama Steven, 1829
Brachyrhinus: sqvamosus: (Fabricius, J.C., 1792)
@gdower (I cannot find colons in the original file. Could it be crawler's problem?). Can we do colon removal before import?

**Comment by mdoering** [comment]
I was wondering if we want to block many names of the same pattern, would it be better to configure the dataset with a  regular expression to match the pattern which blocks all names in that dataset instead of doing individual decisions for each name? We would need a preview of all names that will get blocked I assume, but that isn't very difficult to produce. Sounds much more effective and maintainable than individual decisions.

**Comment by yroskov** [comment]
Agree. Individual decisions have limited application, they doesn't work with many names (>50) in the report.

Excel filters were much more efficient in old dataflow.

**Comment by mdoering** [comment]
I have added pattern regex matching to the API to give you an idea of what we could easily do:

This matches all names that have a German Umlaut in their name incl authorship.
The regex pattern `[öäü]` is `[URL encoded](https://meyerweb.com/eric/tools/dencoder/)`, sth the UI would do for you:

http://api.dev.catalogueoflife.org/dataset/1018/nameusage/pattern?limit=10&regex=%5B%C3%B6%C3%A4%C3%BC%5D

The pattern `^Curculio [0-9]` would match all `Curculio SOME_NUMBER` names:
http://api.dev.catalogueoflife.org/dataset/1039/nameusage/pattern?limit=100&regex=%5ECurculio%20[0-9]

To only match names that have a space or the end of the name followed the number, you can use `^Curculio [0-9]+( |$)`:
http://api.dev.catalogueoflife.org/dataset/1039/nameusage/pattern?limit=100&regex=%5ECurculio%20%5B0-9%5D%2B(%20%7C%24)

Note: You can also use curl to url encode the regex query param for you in the lack of a UI and [jq](https://stedolan.github.io/jq/) to visualize results nicely:
curl -s --data-urlencode "regex=^Curculio [0-9]+( |$)" -G "http://api.dev.catalogueoflife.org/dataset/1039/nameusage/pattern?limit=10" | jq '.[] | (.name + " " + .authorship)'

**Comment by mdoering** [comment]
We could make use of this for the COL assembly:
 - define source dataset (or sector) specific batch decision patterns with a simple action associated, so that each name usage that matches the pattern and a few other optional filters (rank & status) will be blocked or its status updated
 - provide a preview for the UI to list all names that currently match that pattern
 - log all matches in a sync report, so you can review which names have been effectively blocked or updated

From the backend there is not much work to support this, maybe 1-2 days max.
I think this would be really useful to reduce otherwise very repetitive work.

**Comment by yroskov** [comment]
**New view of WTaxa imported 2021-02-25: https://data.catalogueoflife.org/catalogue/3/dataset/1039**

- [ ] Partially Parsable Name, 847->203 & Indetermined, 850->206
A, names such as Curculio 470 Zschach, J.J., 1788 blocked via decisions - fixed
B, colons - fixed by @gdower 
C, doubled subgenus - **not fixed yet**
D, names such as Curculio 2.maculatus  - fixed by @gdower as Curculio 2-maculatus

**Comment by yroskov** [comment]
- [x] Unusual Name Characters, 536
A. Characters with diacritics - left unchanged
B. Names with "?" - left unchanged
C. Names like that Cryptorhis 4tuberculatus converted to Cryptorhis 4-tuberculatus

**Comment by yroskov** [comment]
**WTaxa of 2021-04-04**. Imported on Dev 2021-05-05
https://data.dev.catalogueoflife.org/dataset/2135/classification

There are many duplicated species names.

Few examples of duplicated names:
Batatarhynchus destructor Hustache, 1933
![image](https://user-images.githubusercontent.com/28574252/117195618-19469100-adab-11eb-9fcf-b34b6d9747af.png)

Batames divisus Casey, 1922
![image](https://user-images.githubusercontent.com/28574252/117195695-31b6ab80-adab-11eb-95e1-e04588011be9.png)

Barytychius rufipennis Pic, 1914
![image](https://user-images.githubusercontent.com/28574252/117195777-527f0100-adab-11eb-81f2-154daa433fc2.png)

**Comment by yroskov** [comment]
Misspellings?
![image](https://user-images.githubusercontent.com/28574252/117197271-3b411300-adad-11eb-8223-9d7374b8ecf1.png)

![image](https://user-images.githubusercontent.com/28574252/117197423-6c214800-adad-11eb-9705-73304e138750.png)

**Comment by yroskov** [comment]
**WTaxa of 2021-04-04. Imported on PROD 2021-06-30**
https://data.catalogueoflife.org/catalogue/3/dataset/1039/classification

- [x] Metadata updated
- [x] Genera overlapped with 3i Curculio have been blocked programmatically by Geoff

ISSUES resolved

TASKS resolved 2021-06-30

![image](https://user-images.githubusercontent.com/28574252/124007757-49707180-d9a1-11eb-98d6-364d2a9ebdac.png)

**Synced 2021-06-30**

**Comment by yroskov** [comment]
- [x]  **genus Sitones blocked** in WTaxa 2024-10-07 (it's is a synonym of Entiminae Sitona): https://github.com/CatalogueOfLife/data/issues/706

WTaxa re-synced 2024-10-07

**Comment by yroskov** [comment]
- [x] genus Calendra Clairville, J.P.de] ., 1798 vs Calandra Clairville, J.P.de] ., 1798. 
Calendra (18) = blocked 2025-01-14

WTaxa re-synced 2025-01-14

**Comment by yroskov** [comment]
- [x] WTaxa (id 1039) = 133 duplicated accepted genera with other GSDs = RESOLVED (see spreadsheet), **re-synced 2026-07-13**

Junior homonyms flagged as prov acc; duplicates with Entiminae blocked.
